google-cloud-storage/lib/google/cloud/storage/policy/binding.rb

```diff
@@ -0,0 +1,239 @@
+          #   bindings, including their conditions, are examined independently.
+          #   Optional.
+          #
+          def initialize role:, members:, condition: nil
+            @role = String role
+
+            @members = Array members
+            raise ArgumentError, "members is empty, must be provided" if @members.empty?
+
+            condition = Condition.new(**condition) if condition.is_a? Hash
+            if condition
+              raise ArgumentError, "expected Condition, not #{condition.inspect}" unless condition.is_a? Condition
+            end
+            @condition = condition
+          end
+
+          ##
+          # Sets the role for the binding.
+          #
+          # @param [String] new_role Role that is assigned to members. For example,
+          #   `roles/viewer`, `roles/editor`, or `roles/owner`. Required.
+          #
+          def role= new_role
+            @role = String new_role
+          end
+
+          ##
+          # Sets the members for the binding.
+          #
+          # @param [Array<String>] new_members Specifies the identities requesting
+          #   access for a Cloud Platform resource. members can have the
+          #   following values. Required.
+          #
+          #   * `allUsers`: A special identifier that represents anyone who is on
+          #     the internet; with or without a Google account.
+          #   * `allAuthenticatedUsers`: A special identifier that represents
+          #      anyone who is authenticated with a Google account or a service
+          #      account.
+          #   * `user:{emailid}`: An email address that represents a specific
+          #     Google account. For example, `alice@example.com`.
+          #   * `serviceAccount:{emailid}`: An email address that represents a
+          #     service account. For example, `my-other-app@appspot.gserviceaccount.com`.
+          #   * `group:{emailid}`: An email address that represents a Google group.
+          #     For example, `admins@example.com`.
+          #   * `domain:{domain}`: The G Suite domain (primary) that represents
+          #     all the users of that domain. For example, `google.com` or
+          #     `example.com`. Required.
+          #
+          def members= new_members
+            new_members = Array new_members
+            raise ArgumentError, "members is empty, must be provided" if new_members.empty?
+            @members = new_members
+          end
+
+          ##
+          # Sets the condition for the binding.
+          #
+          # @param [Google::Cloud::Storage::Policy::Condition] new_condition The
+          #   condition that is associated with this binding. NOTE: An unsatisfied
+          #   condition will not allow user access via current binding. Different
+          #   bindings, including their conditions, are examined independently.
+          #   Optional.
+          # @overload new(title:, description: nil, expression:)
```

Comment:
I don't understand this overload. Where does the `new` method come from?

Reply:
Fixed.